src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -154,22 +154,25 @@ std::string HydroProduction::infeasibilityCause()
 // --- Constraints factory ---
 ConstraintsFactory::ConstraintsFactory()
 {
-    regex_to_ctypes_ = {
-      {"::hourly::", [](std::string name) { return std::make_shared<HourlyBC>(name); }},
-      {"::daily::", [](std::string name) { return std::make_shared<DailyBC>(name); }},
-      {"::weekly::", [](std::string name) { return std::make_shared<WeeklyBC>(name); }},
-      {"^FictiveLoads::", [](std::string name) { return std::make_shared<FictitiousLoad>(name); }},
-      {"^AreaHydroLevel::", [](std::string name) { return std::make_shared<HydroLevel>(name); }},
-      {"^Level::", [](std::string name) { return std::make_shared<STS>(name); }},
-      {"^HydroPower::", [](std::string name) { return std::make_shared<HydroProduction>(name); }}};
-}
-
-std::shared_ptr<WatchedConstraint> ConstraintsFactory::create(std::string name) const
-{
-    return std::ranges::find_if(regex_to_ctypes_,
-                                [&name](auto& pair)
-                                { return std::regex_search(name, std::regex(pair.first)); })
-      ->second(name);
+    regex_to_ctypes_ = {{"::hourly::", std::make_unique<HourlyBC, const std::string&>},
+                        {"::daily::", std::make_unique<DailyBC, const std::string&>},
+                        {"::weekly::", std::make_unique<WeeklyBC, const std::string&>},
+                        {"^FictiveLoads::", std::make_unique<FictitiousLoad, const std::string&>},
+                        {"^AreaHydroLevel::", std::make_unique<HydroLevel, const std::string&>},
+                        {"^Level::", std::make_unique<STS, const std::string&>},
+                        {"^HydroPower::", std::make_unique<HydroProduction, const std::string&>}};
+}
+
+std::unique_ptr<WatchedConstraint> ConstraintsFactory::create(const std::string& name) const
```

Comment:
About **ConstraintsFactory::create(...)** :
It seems impossible, from the **name**, not to find the right map value : the **name** necessarily matches a key in the map.
 That's the reason why I didn't introduce the check you just added.

Reply:
ok, if you like it like that. 
At least, if someone adds a new constraint to watch, the problem you point won't happen.

Reply:
Yes, that's the whole point